frame/society/src/lib.rs

```diff
@@ -1492,20 +1631,32 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 					if is_accepted {
 						total_approvals += approval_count;
 						total_payouts += value;
-						members.push(candidate.clone());
 
-						Self::pay_accepted_candidate(&candidate, value, kind, maturity);
+						match kind {
+							BidKind::Deposit(_) | BidKind::Vouch(_, _) => {
+								members.push(candidate.clone());
+							},
+							// Winning action bid does not become member
+							BidKind::Action(_, _) => {},
+						}
+
+						Self::execute_accepted_candidate(&candidate, value, kind, maturity);
 
 						// We track here the total_approvals so that every candidate has a unique
 						// range of numbers from 0 to `total_approvals` with length `approval_count`
 						// so each candidate is proportionally represented when selecting a
 						// "primary" below.
 						Some((candidate, total_approvals, value))
```

Comment:
if we include action bids here (ie, instead of returning None) that means further on (L1726) they are considered as candidates for the new head and passed as new members to T::MembershipChanged - this should have no effect on the logic of this pallet, but may have undesirable effects on the rest of the runtime.

Reply:
Hum...will double check that, but as we do not have a check for existing members here we will have duplication in `<Members<T, I>>`.

Reply:
Replacing the `match kin { .. }` by `members.push(candidate.clone());` will break tests:
```
---- tests::bid_judge_actions_works stdout ----
thread 'tests::bid_judge_actions_works' panicked at 'assertion failed: `(left == right)`
  left: `[10, 10, 30, 50]`,
 right: `[10, 30, 50]`', frame/society/src/tests.rs:1110:9
 ```
As you can see,`10` is duplicated =/